Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -531,6 +528,12 @@ class BrowserViewController: UIViewController {
                 profile.prefs.setString(appVersion, forKey: "latestAppVersion")
             }
         }
+
+        if var queuedAlertInfo = tabManager.selectedTab?.dequeueJavascriptAlertPrompt() {
+            let alertController = queuedAlertInfo.alertController()
+            alertController.delegate = self
```

Comment:
Why do we need this in `viewDidAppear`?

Since this is exactly the same code you have in `promptAlertControllerDidDismiss`, maybe factor this out into a private helper method?